power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_gen/input.hpp

```diff
@@ -0,0 +1,752 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+// This header file is automatically generated. DO NOT modify it manually!
+
+// clang-format off
+#pragma once
+#ifndef POWER_GRID_MODEL_AUXILIARY_META_GEN_INPUT_HPP
+#define POWER_GRID_MODEL_AUXILIARY_META_GEN_INPUT_HPP
+
+#include "../../enum.hpp" // NOLINT
+#include "../../power_grid_model.hpp" // NOLINT
+#include "../../three_phase_tensor.hpp" // NOLINT
+#include "../meta_data.hpp" // NOLINT
+#include "../input.hpp" // NOLINT
+
+
+namespace power_grid_model {
+
+namespace meta_data {
+
+// template specialization functors to get attributes
+
+template<>
+struct get_attributes_list<BaseInput> {
+    std::vector<MetaAttribute> operator() () const {
+        std::vector<MetaAttribute> attributes{};
+        // all attributes including base class
+        
+        attributes.push_back(MetaAttribute(MetaAttributeImpl<BaseInput, &BaseInput::id>{}, "id"));
+        return attributes;
+    }
+};
+
+template<>
+struct get_attributes_list<NodeInput> {
+    std::vector<MetaAttribute> operator() () const {
+        std::vector<MetaAttribute> attributes{};
+        // all attributes including base class
+        
+        attributes.push_back(MetaAttribute(MetaAttributeImpl<NodeInput, &NodeInput::id>{}, "id"));
+        attributes.push_back(MetaAttribute(MetaAttributeImpl<NodeInput, &NodeInput::u_rated>{}, "u_rated"));
+        return attributes;
```

Comment:
I can use `emplace_back` to save some spaces, or just like you suggest directly initialize in the `vector`. Which do you prefer?

`constexpr` is not possible because we have string here. `constexpr std::string` can only be used if the value is not used in run-time.

Reply:
Have used your suggestion